integration/internal/process.go

```diff
@@ -135,3 +142,18 @@ func (ps *ProcessState) Stop() error {
 
 	return nil
 }
+
+func (ps *ProcessState) CopyTo(
```

Comment:
I think that's slightly over-engineered.

Reply:
That's totally true. This is the result of multiple experiments and refactors and somewhere along the way I clearly just have lost ~~my mind~~ what's actually going on :)